compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEAssignNpuDmaBdIds.cpp

```diff
@@ -257,6 +416,144 @@ LogicalResult releaseBdId(
   return success();
 }
 
+/// DmaOps are assigned Bd Ids prior to invoking this function and a map
+/// `dmaOpToBdIdMap` is maintained that maps a DmaOp to its source/target Bd
+/// Ids. For each DmaOp in the list `dmaOps`, this API will check the
+/// `dmaOpToBdIdMap` and release Bd Ids if assigned.
+static LogicalResult releaseAssignedBdIdsInCurrentBatch(
+    SmallVector<AMDAIE::NpuDmaCpyNdOp> dmaOps,
+    DenseMap<Value, ChannelBdIdGenerator> &shimTileToGeneratorMap,
+    DenseMap<AMDAIE::BdIdOp, SmallVector<uint32_t>> &bdIdOpToBdIdsMap,
+    DenseMap<AMDAIE::NpuDmaCpyNdOp, SmallVector<AMDAIE::BdIdOp>>
+        &dmaOpToBdIdMap) {
+  // Release BD ID used by input DMA op.
+  for (AMDAIE::NpuDmaCpyNdOp npuDmaOp : dmaOps) {
+    if (dmaOpToBdIdMap[npuDmaOp][0]) {
```

Comment:
Is this guaranteed that index `[0]` and `[1]` exist in `SmallVector<AMDAIE::BdIdOp>`? I think this is always a pair, so `std::pair` would be more appropriate?

Reply:
Yes, although this is guaranteed to always have index `[0]` and `[1]` because of [this line](https://github.com/nod-ai/iree-amd-aie/blob/f74e7ddb4819e3f17f6064c3e34b00bcf9e6db4f/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEAssignNpuDmaBdIds.cpp#L361) - but using `std::pair` here would render [this one-line update](https://github.com/nod-ai/iree-amd-aie/blob/f74e7ddb4819e3f17f6064c3e34b00bcf9e6db4f/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEAssignNpuDmaBdIds.cpp#L365) useless.

Is there anything else we could consider using here that has the essence of that one-line capture?